### DISCUSSION TITLE
Welcome to predict_Lottery_ticket Discussions!

### DISCUSSION THREAD

**Discussion by KittenCN** [KittenCN/predict_Lottery_ticket#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by laydownfish** [comment]
请问kl8_cahs.py文件怎么使用？提示需要一个现金文件，请问是自己放进去吗？有什么格式要求吗？

**Reply by KittenCN** [reply]
kl8开头的是我用来测试一些公式的，一般不需要使用；kl8_cash是用来计算号码获奖以及返奖率的文件

**Reply by laydownfish** [reply]
谢谢，另外请问window_size参数config里面有，那运行的时候--window_size是否会覆盖config里的？

**Reply by KittenCN** [reply]
--参数 就是为了快速覆盖config的固定参数而使用

**Reply by laydownfish** [reply]
谢谢，另外是否考虑增加验证集验证是否存在过拟合现象？有时候loss过低也有可能是过拟合

**Reply by laydownfish** [reply]
主要是我epoch轮数多了后虽然loss低了很多，但是实际预测还是不准。。。。

**Reply by KittenCN** [reply]
嗯，我也发现了，只是最近的兴趣在区块链上，等我手头的东西写完了，就回来搞这个

**Reply by laydownfish** [reply]
那我先自己试试看看交叉验证法，另外大佬快乐8的参数能方便分享下吗？我想做几组对比验证。

**Reply by KittenCN** [reply]
我没有特殊的参数，都是系统周期性平均值，可以用--calculate_rate加上--calculate_rate_list参数自动计算；其中calculate_rate 为1时表示计算参数；calculate_rate_list 设置选几

**Reply by laydownfish** [reply]
我已经建立了一个简单的Transformer模型，预测方面我用了回测，也就是将20%的历史数据用来测试，没有纳入训练。但是发现效果不理想。想和大佬再交流学习，但是我本身是半路出家的，对python也是现学没多久，家里电脑也很老旧所以代码都在notebook上运行。如果大佬方面留个联系方式的话我把代码发过去。

**Reply by KittenCN** [reply]
可以发到我的邮箱：owdely@gmail.com

**Reply by laydownfish** [reply]
已经发了大佬

**Comment by Forward833** [comment]
請教一下作者：

**Comment by Forward833** [comment]
請教一下作者：我在訓練的過程中出現這樣的信息：
![11111](https://github.com/KittenCN/predict_Lottery_ticket/assets/168695618/c9210df8-60f0-4823-99b8-3c259a1c2af6)
是否會對訓練的合理性有影響？ 另外本方案是否需要日常盡量多的訓練？

**Comment by KittenCN** [comment]
> 請教一下作者：我在訓練的過程中出現這樣的信息： ![11111](https://private-user-images.githubusercontent.com/168695618/327359452-c9210df8-60f0-4823-99b8-3c259a1c2af6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2OTk1MzcsIm5iZiI6MTcxNDY5OTIzNywicGF0aCI6Ii8xNjg2OTU2MTgvMzI3MzU5NDUyLWM5MjEwZGY4LTYwZjAtNDgyMy05OWI4LTNjMjU5YTFjMmFmNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwM1QwMTIwMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NmI1MzJkY2Q3OGRlM2MxYmZjYTQxZDUyMGJjOTY5NDExZTZmMzM0YjMwYmExZGVkMzdiNjBiM2Y5NmY5OWEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ob1YYiKaXbuehZXA4zCYtzi2lwD9c7CXrZATxZYr0mE) 是否會對訓練的合理性有影響？ 另外本方案是否需要日常盡量多的訓練？

你的使用环境和参数能提供下吗？只要loss下降到合理的地步，就不需要一直训练了，日后有新的数据加入，再训练

**Reply by Forward833** [reply]
比如雙色球是每兩天更新一組數據，是否每次更新時重新get_data一次，然後訓練到loss下降到合理的地步？（冒昧問一下loss下降到什麽程度為之合理？）
我用的電腦比較舊： i5-9400F ＋ GTX1050ti ＋ 16GRAM；Win10 專業版（19045.3930）；SSD也比較久：KINGSTON SA400S37480G（500GB）；但SSD分了C、D兩個分區，系統在C；本方案所有程序、文件都在C盤但因為空間不夠，oneAPI安裝在D分區（不知是否有影響）
我的requirements：
tensorflow-cpu==2.3.*
# tensorflow-gpu==2.3
tensorflow-addons==0.13.*
requests
keras
beautifulsoup4
pandas==1.0.0
numpy==1.16.0
lxml
loguru
grpcio
protobuf
其中numpy是安裝過程中按照提示設為1.16.0；也把protobuf重装降至3.20.1

**Comment by Forward833** [comment]
我發現好像紅球模型訓練到loss在40％左右就不能再降低了，會在42％～47％之間徘徊，籃球模型則是1.03％～1.5％之間徘徊，籃球應該足夠低了但紅球顯然應該是不夠低的，是否需要加大batch_size？如果加大之後在預測之前再次需要用batch_size＝1再訓練一次才能預測的話，是否會互相影響？

**Reply by KittenCN** [reply]
这个项目我有短时间没碰了，我记忆里面双色球下，红蓝都可以降到10%以下，但是需要极大量的训练，以及调整参数，我记得我训练了至少3-5天

**Reply by Forward833** [reply]
我也是不斷的訓練，現在籃球已經降到0.7％上下了，而紅球依然在40％上下徘徊～

**Reply by KittenCN** [reply]
增加学习率试试

**Comment by Forward833** [comment]
現在訓練到紅球20％了，但是出現一種情況：執行訓練一開始“創建數據集”這個步驟有一定幾率停滯（程序並沒有卡死，游標還在閃爍），有時候等待一定時間后能自動進行下去，但更多時候時一直停滯，要關掉程序。不知是否會影響訓練效果？

**Comment by Forward833** [comment]
紅球始終降不到10％以內  作者能否分享一下您訓練過程認為合適的epoch值和batch_size值？（本人對預測結果能否命中並不執著 本人是初學者 出於學習同時給生活一點期待值而已 ）

**Reply by KittenCN** [reply]
我也忘记我的参数了，这个项目已经很老了，用的技术也很老，如果是为了学习，建议去我另一个pytorch的版本，当然，那个版本目前只支持快乐8，但是其基本结构都已经完整了，只要修改数据的结构，就应该能支持其他玩法。

**Reply by Forward833** [reply]
有留意這個項目，但快8是需要高投入的，本人初學還不能自己通過修改到SSQ，還是等大神您進展到那個階段～

**Reply by KittenCN** [reply]
我目前手头有两个非常重要的公司项目要做，短时间不太可能来更新这些东西了。

**Comment by Forward833** [comment]
請教一下作者 理論上在某個epoch值下訓練完成後，如果進行下一次訓練，不管epoch值是和前一次相同還是不同的值，後一次好像是不會繼承前一次學習的成果來迭代的，這樣每一期的數據更新之後  是不是都要進行一次epoch3000～5000的訓練？

**Comment by Forward833** [comment]
很奇怪，試驗了兩週，發現如果將epochs設到4000～5000，完成數次訓練（每次花費4天左右）後，紅球的loss還是無法降到10％以內，但這樣訓練後執行預測出來的紅號全部都非常集中，比如：22 23 24 26 28 29、14 16 19 20 21 23、12 13 14 15 27 28 29等等，和之前以較低的epochs訓練的預測有很大區別，之前偶爾也會有集中的情況但不會每一組都這麼集中。當然這些組合也是可能的組合之一，但當只會預測出這種集中結果時，是否代表這個模型已經出了某種問題？
